### PR TITLE
Fix some warnings in Emby.Server.Implementations

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

File: Emby.Server.Implementations/EntryPoints/RecordingNotifier.cs [hunk 1]
```diff
@@ -30,30 +30,30 @@ public RecordingNotifier(ISessionManager sessionManager, IUserManager userManage
 
         public Task RunAsync()
         {
-            _liveTvManager.TimerCancelled += _liveTvManager_TimerCancelled;
-            _liveTvManager.SeriesTimerCancelled += _liveTvManager_SeriesTimerCancelled;
-            _liveTvManager.TimerCreated += _liveTvManager_TimerCreated;
-            _liveTvManager.SeriesTimerCreated += _liveTvManager_SeriesTimerCreated;
+            _liveTvManager.TimerCancelled += LiveTvManagerTimerCancelled;
+            _liveTvManager.SeriesTimerCancelled += LiveTvManagerSeriesTimerCancelled;
```
Comment (dkanada) [inline Emby.Server.Implementations/EntryPoints/RecordingNotifier.cs]: I would add `On` to the front here to match the others.

File: Emby.Server.Implementations/LiveTv/LiveTvMediaSourceProvider.cs [hunk 2]
```diff
@@ -100,11 +92,13 @@ private async Task<IEnumerable<MediaSourceInfo>> GetMediaSourcesInternal(BaseIte
...
-                    openKeys.Add(item.Id.ToString("N", CultureInfo.InvariantCulture));
-                    openKeys.Add(source.Id ?? string.Empty);
-                    source.OpenToken = string.Join(StreamIdDelimeterString, openKeys.ToArray());
+                    var openKeys = new List<string>
+                    {
+                        item.GetType().Name,
+                        item.Id.ToString("N", CultureInfo.InvariantCulture),
+                        source.Id ?? string.Empty
+                    };
+                    source.OpenToken = string.Join(StreamIdDelimeterString, openKeys);
```
Comment (dkanada) [inline Emby.Server.Implementations/LiveTv/LiveTvMediaSourceProvider.cs]:
```suggestion

                    source.OpenToken = string.Join(StreamIdDelimeterString, openKeys);
```

File: Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs [hunk 3]
```diff
@@ -97,10 +99,10 @@ private void UpdateTimerCallback(object state)
 
                 var task = SendNotifications(changes, CancellationToken.None);
 
-                if (UpdateTimer != null)
+                if (_updateTimer != null)
                 {
-                    UpdateTimer.Dispose();
-                    UpdateTimer = null;
+                    _updateTimer.Dispose();
+                    _updateTimer = null;
```
Comment (JustAMan) [inline Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs]:
```c#
_updateTimer?.Dispose();
_updateTimer = null;
```
?

File: Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs [hunk 4]
```diff
@@ -145,13 +147,13 @@ private UserDataChangeInfo GetUserDataChangeInfo(Guid userId, List<BaseItem> cha
...
         public void Dispose()
         {
-            if (UpdateTimer != null)
+            if (_updateTimer != null)
             {
-                UpdateTimer.Dispose();
-                UpdateTimer = null;
+                _updateTimer.Dispose();
+                _updateTimer = null;
             }
```
Comment (JustAMan) [inline Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs]:
```c#
_updateTimer?.Dispose();
_updateTimer = null;
```
?

File: Emby.Server.Implementations/HttpServer/HttpListenerHost.cs [hunk 5]
```diff
@@ -82,16 +84,14 @@ public class HttpListenerHost : IHttpServer, IDisposable
...
-
         public object CreateInstance(Type type)
         {
             return _appHost.CreateInstance(type);
         }
 
         private static string NormalizeUrlPath(string path)
         {
-            if (path.StartsWith("/"))
+            if (path.Length > 0 && path[0] == '/')
```
Comment (JustAMan) [inline Emby.Server.Implementations/HttpServer/HttpListenerHost.cs]: wtf, this isn't a full normalization... :) doesn't asp.net have a normalization routine? I personally prefer slow but correct (and secure) code to fast but incorrect 😛 

cc @Artiume @cvium

Comment (JustAMan) [reply Emby.Server.Implementations/HttpServer/HttpListenerHost.cs]: Note that normalization should probably be fixed in a separate PR, so it is not a blocker now.

File: Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs (hunk 4 above)
Comment (Bond-009) [reply Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs]: Wht? We check the line before if it's not null.

File: Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs (hunk 3 above)
Comment (Bond-009) [reply Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs]: Why? We check the line before if it's not null.

File: Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs (hunk 4 above)
Comment (JustAMan) [reply Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs]: because it's shorter? Idc much tho

Comment (Bond-009) [reply Emby.Server.Implementations/EntryPoints/UserDataChangeNotifier.cs]: Oh, yeah we could...
Doesn't really matter tho, does it?

**Pull request merged**